AirGradient-DIY.ino

```diff
@@ -1,117 +1,161 @@
-
-Compatible with the following sensors:
-Plantower PMS5003 (Fine Particle Sensor)
-SenseAir S8 (CO2 Sensor)
-SHT30/31 (Temperature/Humidity Sensor)
-
-Please install ESP8266 board manager (tested with version 3.0.0)
-
-The codes needs the following libraries installed:
-"WifiManager by tzapu, tablatronix" tested with Version 2.0.3-alpha
-"ESP8266 and ESP32 OLED driver for SSD1306 displays by ThingPulse, Fabrice Weinberg" tested with Version 4.1.0
-
-Configuration:
-Please set in the code below which sensor you are using and if you want to connect it to WiFi.
-
-If you are a school or university contact us for a free trial on the AirGradient platform.
-https://www.airgradient.com/schools/
-
-MIT License
-*/
+/**
+ * This is good enough if not better. It is simple and works. So it shall be.
+ */
 
+// 
 #include <AirGradient.h>
-#include <WiFiManager.h>
 #include <ESP8266WiFi.h>
-#include <ESP8266HTTPClient.h>
+#include <ESP8266WebServer.h>
+#include <WiFiClient.h>
 
 #include <Wire.h>
 #include "SSD1306Wire.h"
 
 AirGradient ag = AirGradient();
 
-SSD1306Wire display(0x3c, SDA, SCL);
+//
+// Config Start
+
+const char* deviceId = "AirGradient";
+
+const bool hasPM = true;
+const bool hasCO2 = true;
+const bool hasSHT = true;
 
-// set sensors that you do not use to false
-boolean hasPM=true;
-boolean hasCO2=true;
-boolean hasSHT=true;
+const char* ssid = "PleaseChangeMe";
+const char* password = "PleaseChangeMe";
 
-// set to true if you want to connect to wifi. The display will show values only when the sensor has wifi connection
-boolean connectWIFI=true;
+const int port = 9926;
 
-// change if you want to send the data to another server
-String APIROOT = "http://geerli.net:9926/";
+long lastUpdate;
+//The frequency of measurement updates
+const int updateFrequency = 5000; 
 
-void setup(){
+int counter = 0;
+
+// Config End
+//
+SSD1306Wire display(0x3c, SDA, SCL);
+ESP8266WebServer server(port);
+//
+
+void setup() {
   Serial.begin(9600);
 
+  // Init Display.
   display.init();
   display.flipScreenVertically();
   showTextRectangle("Init", String(ESP.getChipId(),HEX),true);
 
+  // Enable enabled sensors.
   if (hasPM) ag.PMS_Init();
   if (hasCO2) ag.CO2_Init();
   if (hasSHT) ag.TMP_RH_Init(0x44);
 
-  if (connectWIFI) {
-    showTextRectangle("Connect", "to WiFi",true);    
-    connectToWifi();
+  // Setup and wait for wifi.
+  WiFi.begin(ssid, password);
+  Serial.println("");
+  while (WiFi.status() != WL_CONNECTED) {
+    delay(500);
+    Serial.print(".");
   }
-  delay(2000);
+
+  Serial.println("");
+  Serial.print("Connected to ");
+  Serial.println(ssid);
+  Serial.print("IP address: ");
+  Serial.println(WiFi.localIP());
+  Serial.print("MAC address: ");
+  Serial.println(WiFi.macAddress());
+
+  server.on("/", HandleRoot);
+  server.on("/metrics", HandleRoot);
+  server.onNotFound(HandleNotFound);
+
+  server.begin();
+  Serial.println("HTTP server started at ip " + WiFi.localIP().toString() );
```

Comment:
Humbly suggest including the port to this Serial message

Reply:
I added it to both the display and serial now.